controllers/mongodb_users.go

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+	"fmt"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/secret"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity) error {
+	for _, user := range mdb.GetScramUsers() {
+		secretNamespacedName := types.NamespacedName{Name: user.PasswordSecretName, Namespace: mdb.NamespacedName().Namespace}
+		if _, err := secret.ReadKey(r.client, user.PasswordSecretKey, secretNamespacedName); err != nil {
+			if apiErrors.IsNotFound(err) {
+				r.log.Errorf(`User password secret "%s" not found`, secretNamespacedName)
```

Comment:
I'm not sure about logging here, we will return the error anyway at we will both see an error and a log message?

What do you think if we returned fmt.Errorf(...) here?

Reply:
done